deploy-board/deploy_board/webapp/env_views.py

```diff
@@ -862,20 +830,16 @@ def post_add_stage(request, name):
     stages, _ = common.get_all_stages(all_envs_stages, None)
     if from_stage and from_stage not in stages:
         raise Exception("Can not clone from non-existing stage!")
-    
-    external_id = None
-    if IS_PINTEREST:
-        identifier = create_identifier_for_new_stage(request, name, stage)
-        external_id = identifier.get('uuid') if not identifier == None else None # if there is no stage in this env with externalId, still create the new stage
+
+    external_id = environs_helper.create_identifier_for_new_stage(request, name, stage)
 
     try:
         if from_stage:
             common.clone_from_stage_name(request, name, stage, name, from_stage, description, external_id)
         else:
             common.create_simple_stage(request,name, stage, description, external_id)
     except:
-        if IS_PINTEREST:
-            environs_helper.delete_nimbus_identifier(request, external_id)
+        environs_helper.delete_nimbus_identifier(request, external_id)
```

Comment:
If there is any bad data, will this delete the nimbus identifier causing an incident for old stages?

Reply:
looks like external ID can only be set if new, so i think this one is OK